scripts/09_deploy_openshift.sh

```diff
@@ -12,13 +12,39 @@ mkdir -p ocp/openshift
 python3 /root/bin/ipmi.py off
 python3 /root/bin/redfish.py off
 {% if bmc_reset %}
-  {% for worker in workers %}
-  {% if worker['model']|default('kvm') == "dell" %}
-      worker_ip={{ worker["redfish_address"] }}
-      curl -i -k -X POST -H 'Content-Type: application/json' -H 'Accept: application/json' -u {{ bmc_user }}:{{ bmc_password }} --data '{"ResetType":"GracefulRestart"}' 'https://"${worker_ip}"/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Manager.Reset'
-    {% endif %}
-  {% endfor %}
+{% for worker in workers %}
+{% if worker['model']|default('kvm') == "dell" %}
+worker_ip={{ worker["redfish_address"] }}
+curl -i -k -X POST -H "Content-Type: application/json" -H "Accept: application/json" -u root:calvin --data '{"ResetType":"GracefulRestart"}' "https://${worker_ip}/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Manager.Reset"
+if [[ $? -eq 0 ]]; then  # Can be implicit, but explicit for visibility
+  echo "BMC ${worker_ip} restarting!"
+  sleep 30  # Time to start restart
+else
+  echo "ERROR: BMC ${worker_ip} fails to restart"
+  # exit 1
+fi
+{% endif %}
+{% endfor %}
+{% endif %}
+
+{% for worker in workers %}
+{% if worker['model']|default('kvm') == "dell" %}
+worker_ip={{ worker["redfish_address"] }}
+SECONDS_PASSED=0
+SECONDS_TIMEOUT=300
+until curl -k https://${worker_ip}/redfish/v1/ > /dev/null
+do
+  echo "Waiting for worker ${worker_ip} to come back up after bmc reset..."
+  sleep 15
+  SECONDS_PASSED=$((SECONDS_PASSED + 5))
```

Comment:
u sleep 15 but add only 5?!
I think it's beter to have SLEEP_INTERVAL
then use it in both places.

Reply:
Yes, you are right. I'll add the SLEEP_INTERVAL variable.